services/src/main/java/org/keycloak/services/resources/account/AccountRestService.java

```diff
@@ -275,42 +259,53 @@ public Response getApplications() {
     @Produces(MediaType.APPLICATION_JSON)
     public Response getApplication(final @PathParam("clientId") String clientId) {
         checkAccountApiEnabled();
-        auth.require(AccountRoles.VIEW_APPLICATIONS);
+        auth.requireOneOf(AccountRoles.MANAGE_ACCOUNT, AccountRoles.VIEW_APPLICATIONS);
         ClientModel client = realm.getClientByClientId(clientId);
-        if (client == null) {
+        if (client == null || client.isBearerOnly() || client.getBaseUrl() == null) {
             return Cors.add(request, Response.status(Response.Status.NOT_FOUND).entity("No client with clientId: " + clientId + " found.")).build();
         }
 
-        return Cors.add(request, Response.ok(modelToRepresentation(client))).build();
+        List<String> inUseClients = new LinkedList<>();
+        if(!session.sessions().getUserSessions(realm, client).isEmpty()) {
+            inUseClients.add(clientId);
+        }
+
+        List<String> offlineClients = new LinkedList<>();
+        if(session.sessions().getOfflineSessionsCount(realm, client) > 0) {
+            offlineClients.add(clientId);
+        }
+
+        return Cors.add(request, Response.ok(modelToRepresentation(client, inUseClients, offlineClients))).build();
     }
 
-    private ClientRepresentation modelToRepresentation(ClientModel model) {
+    private ClientRepresentation modelToRepresentation(ClientModel model, List<String> inUseClients, List<String> offlineClients) {
         ClientRepresentation representation = new ClientRepresentation();
         representation.setClientId(model.getClientId());
-        representation.setClientName(getTranslationOrDefault(model.getName()));
+        representation.setClientName(StringPropertyReplacer.replaceProperties(model.getName(), getProperties()));
+        representation.setDescription(model.getDescription());
+        representation.setUserConsentRequired(model.isConsentRequired());
+        representation.setInUse(inUseClients.contains(model.getClientId()));
+        representation.setOfflineAccess(offlineClients.contains(model.getClientId()));
+        representation.setBaseUrl(model.getBaseUrl());
+        UserConsentModel consentModel = session.users().getConsentByClient(realm, user.getId(), model.getId());
```

Comment:
When you are fetching a list of clients you are already fetching the consent-model, which is being fetched again here. Would be better to not do that.

Reply:
I've updated the PR to pass the consents into modelToRepresentation